presentation/src/main/java/com/whyranoid/presentation/screens/community/SearchFriendScreen.kt

```diff
@@ -153,7 +155,37 @@ fun SearchFriendScreen(
                         onClickItem = { user ->
                             navController.navigate("userPage/${user.uid}/${user.nickname}/${item.isFollowing}")
                         },
-                    )
+                    ) {
+                        if (item.isFollowing) {
+                            Box(modifier = Modifier
+                                .clip(RoundedCornerShape(10.dp))
+                                .background(Color(0xFFE4E4E4))
+                                .clickable {
+                                    viewModel.unFollow(item.user)
+                                }
+                                .padding(horizontal = 12.dp, vertical = 5.dp)) {
+                                Text(
+                                    text = "팔로잉",
+                                    style = WalkieTypography.Body2,
+                                )
+                            }
+                        } else {
+                            Box(modifier = Modifier
+                                .clip(RoundedCornerShape(10.dp))
+                                .background(WalkieColor.Primary)
+                                .clickable {
+                                    viewModel.follow(item.user)
+                                }
+                                .padding(horizontal = 12.dp, vertical = 5.dp)) {
+                                Text(
+                                    text = "팔로우",
+                                    style = WalkieTypography.Body2.copy(
+                                        Color.White
+                                    ),
+                                )
+                            }
+                        }
+                    }
```

Comment:
색상과 문자열 빼곤 컴포넌트의 모양과 액션이 동일하기 때문에 별도의 컴포저블로 분리하는게 어떨까 싶습니다 😎

```suggestion

@Composable
private fun followButton(
    modifier: Modifier = Modifier,
    isFollowing: Boolean,
    onClick: () -> Unit,
) {
    Box(
        modifier = modifier
             .clip(RoundedCornerShape(10.dp))
             .background(if(isFollowing) Color(0xFFE4E4E4) else WalkieTypography.Body2)
             .padding(horizontal = 12.dp, vertical = 6.dp)
             .clickable(onClick = onClick),
) {
    Text(
        text = if (isFollowing) "팔로잉" else "팔로우",
        style = WalkieTypography.Body2,
        color = if(isFollowing) Color.Black else Color.White,
    )
}
```

Reply:
@mangbaam 
오예~ 명범님 감사합니다~~! ㅋㅋㅋㅋㅋㅋㅋ